android/src/main/java/io/invertase/googlemobileads/ReactNativeGoogleMobileAdsCommon.java

```diff
@@ -51,8 +52,16 @@ static AdSize getAdSizeForAdaptiveBanner(String preDefinedAdSize, ViewGroup reac
       DisplayMetrics outMetrics = new DisplayMetrics();
       display.getMetrics(outMetrics);
       int adWidth = (int) (outMetrics.widthPixels / outMetrics.density);
+      float maxAdHeight = ((ReactNativeAdView)reactViewGroup).getMaxAdHeight();
 
       if ("INLINE_ADAPTIVE_BANNER".equals(preDefinedAdSize)) {
+        if (maxAdHeight > 0) {
+          if (maxAdHeight < 32) {
+            return AdSize.getInlineAdaptiveBannerAdSize(adWidth, 32);
+          } else {
+            return AdSize.getInlineAdaptiveBannerAdSize(adWidth, Math.round(maxAdHeight));
+          }
+        }
```

Comment:
I believe you can do something like:
```suggestion
    if (maxAdHeight > 0) {
        return AdSize.getInlineAdaptiveBannerAdSize(adWidth, Math.max(maxAdHeight, 32));
    }
    return AdSize.getCurrentOrientationInlineAdaptiveBannerAdSize(
        reactViewGroup.getContext(), adWidth);
```